src/main/assets/js/opening-hours.ts

```diff
@@ -0,0 +1,88 @@
+import $ from 'jquery';
+
+const formId = '#openingTimesForm';
+const tabId = '#openingTimesTab';
+const newOpeningTimeHeadingId = '#newOpeningHoursHeading';
+const openingTimesContentId = '#openingTimesContent';
+const deleteBtnClass = 'deleteOpeningTime';
+const addOpeningTimesBtnName = 'addOpeningTime';
+const typeSelectName = 'type_id';
+const hoursInputName = 'hours';
+
+const getInputName = (name: string, index: number): string => {
+  return `opening_times[${index}][${name}]`;
+};
+
+// Rename the input fields so that the index values are in order,
+// which affects the order when the form is posted.
+const renameFormElements = (): void => {
+  $(`${tabId} select[name$="[${typeSelectName}]"]`)
+    .attr('name', idx => getInputName(typeSelectName, idx))
+    .attr('id', idx => 'description-' + idx);
+  $(`${tabId} input[name$="[${hoursInputName}]"]`)
+    .attr('name', idx => getInputName(hoursInputName, idx))
+    .attr('id', idx => 'hours-' + idx);
+};
+
+$(() => {
+  if($(tabId).length > 0) {
+
+    // GET the opening-hours data for a court
+    const slug = $('#slug').val();
+    $.ajax({
+      url: `/courts/${slug}/opening-times`,
+      method: 'get',
+      success: (res) => {
+        $(openingTimesContentId).html(res);
+      },
+      error: (jqxhr, errorTextStatus, err) =>
+        console.log(`GET opening times failed: ${errorTextStatus} ${err}`)
+    });
+
+    // POST the opening-hours form for a court
+    $(formId).on('submit', e => {
+      e.preventDefault();
+
+      const url = $(e.target).attr('action');
+      $.post(url, $(e.target).serialize())
+        .done(res => {
+          $(openingTimesContentId).html(res);
+          window.scrollTo(0,0);
+        })
+        .fail(response =>
+          console.log(`POST opening times failed: ${response.status} ${response.responseText}`));
```

Comment:
Just to double check on this one if you wouldn't mind. 

If it fails to connect because the VPN isn't on, or because we are an invalid user, the console.log won't display the name of the IDAM location right? 

I sometimes see logs in the backend on a failure (of the main application) with stuff like "cant connect to blabla.idam.domain.thingy". Which might be bad if it also gets displayed on the frontend too as a log

Reply:
I've updated the code and just removed the 'response' properties to avoid any such issues.